documentation/user-consent.md

```diff
@@ -48,122 +43,269 @@ The document details aspects regarding the consent management, which includes fo
+The API invoker has to provide in the authorization request (/bc_authorize) a login_hint with a valid user identifier together with the credentials and indicate the purpose for accessing the data (Step 1): 
 
-...
+- One option for the identifier is the public IP and (optionally – when applicable) port of the application. Other options could be the MSISDN or other identifiers. NOTE: In IoT scenarios or, in general, in those cases where the consumption device is different than the authorization device, the IP and port is the one of the consumption device for which the network capabilities will be requested/applied.
+- The login_hint is a hint regarding the user for whom authentication is being requested. It has the format `<type>:<identifier>`, for instance, "ipport:" for IP addresses or "tel:+" for phone numbers. A login_hint example for MSISDN "+346xxxyyyzzz" looks like:
 
-## Consent API
+    ```
+    POST /bc-authorize HTTP/1.1
+    Authorization: Basic {Credentials}
+    Content-Type: application/x-www-form-urlencoded
 
->[TO BE EDITED/COMPLETED] - Consent API: API to manage user consent, if required. For example when consent is captured by a third party and stored in operator. A third-party can register the consent captured from a user for a specific purpose.
+    login_hint=tel%3A%2B346xxyyyzzz&
+    … 
+    ```
+- Purpose under which the personal data associated to API consumption will be processed.
 
-## Annex: Flow source code
+The operator OGW platform will:
+
+- Validate user identifier, map it to a telco identifier if applicable, e.g.: map IP to MSISDN. Set the OAuth sub to the unique user id in the operator (Step 2).
+
+- Check if a user consent is needed, which depends on the legal basis associated to the purpose (“legitimate interest”, “contract”, “consent”, etc). If needed, it will check in operator consent master if user consent was already given for that identifier and for the requested purpose(s) (Steps 3-4).
+
+Then, two alternatives may occur.
+
+**Scenario 1**: User consent is not needed or consent is already granted (Step 5). OGW platform will directly return a 200 OK response with the CIBA auth request identifier (auth_req_id=OperatorAuthReqId) to the API invoker. This is a unique identifier to identify the authentication request made by the invoker.
+
+**Scenario 2**: Consent is needed and is not granted by the user yet (Step 6)
+
+- A mechanism is triggered to capture user consent in the operator:
+  
+  - The operator triggers an out-of-band consent capture mechanism to interact with the user. **Operators can choose the consent capture mechanism that best suits their capabilities, preferences and needs**. This mechanism can be a push notification, an SMS, etc.
+  - In parallel, the OGW platform operator returns a 200 OK response with the CIBA auth request identifier (auth_req_id=OperatorAuthReqId) to the API invoker to indicate that the authentication request has been accepted and is going to be processed (Step 6).
```

Comment:
```suggestion
  - In parallel, the Operator's API Gateway Platform returns a 200 OK response with the CIBA auth request identifier (auth_req_id=OperatorAuthReqId) to the API invoker to indicate that the authentication request has been accepted and is going to be processed (Step 6).
```

Reply:
Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/54/commits/c7f9715fa8c597e4c9f94e8f0bbf3aecc501ead5.